src/Raven.Server/Documents/PeriodicBackup/BackupHistory/BackupHistoryStorage.cs

```diff
@@ -0,0 +1,612 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.Backups;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents.TransactionMerger.Commands;
+using Raven.Server.Json;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Logging;
+using Sparrow.Server;
+using Voron;
+using Voron.Data.Tables;
+using Voron.Impl;
+using Bits = Sparrow.Binary.Bits;
+
+namespace Raven.Server.Documents.PeriodicBackup.BackupHistory;
+
+public class BackupHistoryStorage
+{
+    private const string JsonDocumentId = "backup-history-entry";
+
+    private StorageEnvironment _environment;
+    private TransactionContextPool _contextPool;
+
+    private readonly Logger _logger = LoggingSource.Instance.GetLogger<BackupHistoryStorage>("Server");
+
+    private static readonly TableSchema BackupHistoryTableSchema = new();
+    private static readonly TableSchema BackupResultDetailsTableSchema = new();
+
+    private static readonly Slice ByFullBackupIdSlice;
+    private static readonly Slice ByBackupKindSlice;
+
+    static BackupHistoryStorage()
+    {
+        using (StorageEnvironment.GetStaticContext(out var ctx))
+        {
+            Slice.From(ctx, "by-backup-kind", out ByBackupKindSlice);
+            Slice.From(ctx, "by-full-backup-id", out ByFullBackupIdSlice);
```

Comment:
Please check what naming convention we use in other places for index names but I think we should rather use "ByBackupKind" for consistency.